### PR TITLE
Add None placeholder

### PR DESCRIPTION
**Pull request by ahuang11**
Continues addressing https://github.com/holoviz/hvplot/issues/1190
image

### CODE REVIEW

**Comment by Hoxbro** [comment]
Maybe rename `null` to something else. `null` can be confusing to Python-only developers.

**Comment by ahuang11** [comment]
`<None>`?

**Comment by Hoxbro** [comment]
>  I kind of like 

Don't keep me in suspense :upside_down_face:

**Comment by ahuang11** [comment]
Haha, apparently `<no selection>` is hidden without the backticks!

File: hvplot/ui.py [hunk 1]
```diff
@@ -828,6 +847,7 @@ def _populate(self):
                 elif pname == 'groupby' and len(getattr(self, pname, [])) == 0 and len(p.objects) > 2:
                     setattr(self, pname, p.objects[2:])
 
+                p.objects = [NONE_PLACEHOLDER] + p.objects
```
Comment (ahuang11) [inline hvplot/ui.py:850]:
```suggestion                if pname not in ('x', 'y'):
                    p.objects = [NONE_PLACEHOLDER] + p.objects
```

**Comment by maximlt** [comment]
I see the issue says:

> Null value isn't visible in a Multiselect widget, so once a value has been selected, it's not intuitive how to go back to unselected (which is Command-click on Mac)

But is this really true? A select HTML element with multiple options is a very basic element that is used, everywhere?

**Comment by ahuang11** [comment]
I think it's nice to have a <null selection> or empty slot; it's probably more straightforward for the user.

@jbednar was the one who raised this issue so he might be able to provide additional feedback.

**Comment by maximlt** [comment]
I very much like the approach you took when developing the Chat components in Panel, drawing heavily on ChatGPT's design and UX. Let's do the same thing with the explorer? We can pick a champion alternative, or a couple, and gradually adapt the explorer's design to follow theirs. What do you think? Without someone with a UX hat, I find difficult to come to a conclusion on this kind of change.

@jbednar if you think that the MultiSelect widget needs a _null_ option, I suggest a feature request should be opened on Bokeh, as there's nothing specific to the explorer here.

Generally, we need better multi select/choice widgets, this has been in the back of my mind for a while, I'll have to scratch that itch at some point!

**Comment by philippjfr** [comment]
I dislike a null option on a `MultiSelect` widget quite a bit. It means you can have the nonsensical combination of the null value and other values, which is plain weird. The real problem is providing a clear UI for clearing a MultiSelect widget which should indeed happen either in Bokeh or Panel.

**Comment by philippjfr** [comment]
Okay, reading the implementation I see now that you prevent the nonsensical combinations, but having to do this in Python really should not be necessary and is still weird UX. Better multi-select in Bokeh/Panel is the correct way forward, but if you want to merge this as a stopgap I won't object.

**Comment by ahuang11** [comment]
I also vote to add it to Panel/Bokeh, but I think we would need to understand what a better multi-select means--perhaps finding a website that does it really well as Maxime mentioned.

**Comment by philippjfr** [comment]
>To me this is extremely unintuitive behavior, getting the user stuck and unable to back out. I would argue that forcing users to figure out that (on a Mac) they can hold down fn while clicking to unselect is bizarre and hostile UX. I have no idea what's at fault (the browser, Bokeh, Panel, hvPlot??).

Standard behavior of the `<select multiple>` tag: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select_multiple

**Comment by jbednar** [comment]
Still hostile UX! :-)

**Comment by maximlt** [comment]
The UX of the standard multi-select HTML element might be considered bad (I don't think so, YMMV), but I don't think adding another behavior on top of it improves it.

---

There are all sorts of flavors of multi-select widgets out there. There's already a feature request opened on Panel for one of them inspired by Excel https://github.com/holoviz/panel/issues/3018.

**Comment by jbednar** [comment]
> The UX of the standard multi-select HTML element might be considered bad (I don't think so, YMMV), but I don't think adding another behavior on top of it improves it.

Hmm. Going back to it, I see that I'm not even able to detect that it's a multi-select at all in casual usage! Totally hostile UX; how would someone be expected to realize that? Clicking around, it only seems like one item can be selected, and for some reason, after selecting anything, there's no obvious way to unselect it. Yet the actual affordance of this widget is not only to allow multiple selections, it allows no selection too, but only if you discover the magic modifier key. Ugh!!! I hate it. So never mind; it's beyond saving.

**Comment by ahuang11** [comment]
Closing this until we find a better resolution.

**Pull request closed without merging**